### ISSUE TITLE
Unused function parameter only for after-used functions (not for before used)

### ISSUE THREAD

**Issue by kkmuffme** [sirbrillig/phpcs-variable-analysis#57]
let's say I have this: (this is a Wordpress specific example, but can happen anywhere, but since it's so prominently used in WP on every action/filter it helps outline the significance of this)

```
add_action( 'delete_post_meta', 'check_thumbnail_updated_post_meta', -1000, 3 );
function check_thumbnail_updated_post_meta( $meta_id, $post_id, $meta_key ) {
    echo $post_id;
    echo $meta_key;
}
```

Eslint has a handling for this (no-unused-vars with after-use, you mention this here: https://github.com/sirbrillig/phpcs-variable-analysis/issues/21#issuecomment-359600766) which would be exactly what we also would need here in this php check.
So basically if any of the variables is used, then it can discard previous errors for unused vars in this call.

**Comment by sirbrillig** [comment]
Thanks for the report! I think I had meant to do this earlier but forgot. I've started work on this in #58

**Comment by kkmuffme** [comment]
Thanks, however I think "true" should be the default value (as it is in eslint), since usually if later args are used that there are external constraints causing us to not use before args.

**Comment by sirbrillig** [comment]
That's a good point. I was thinking that it's better to make something that ignores warnings an opt-in behavior, but I realized that it's probably pretty rare that these particular warnings are useful.

**Comment by sirbrillig** [comment]
#59 should take care of this.
